server/controllers/middlewares.js

```diff
@@ -95,8 +95,15 @@ export const getOrCreateUser = (req, res, next) => {
 
   // If already logged in, proceed
   if (req.remoteUser) {
-    req.user = req.remoteUser;
-    return next();
+    if (req.body.expense) {
+      const {name, paypalEmail} = req.body.expense;
+      return req.remoteUser.updateWhiteListedAttributes({ name, paypalEmail })
```

Comment:
not a fan of adding this in the `getOrCreateUser`. It doesn't seem to be the right place.
Also, for more visibility would be good to add a check 
`if (paypalEmail !== req.remoteUser.paypalEmail)`